### PR TITLE
[Benchmark] update node classification for multigpu benchmark

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

literally copy from https://github.com/dmlc/dgl/blob/master/examples/pytorch/multigpu/multi_gpu_node_classification.py with a few arguments including `dataset_name`, `dataset_dir`, `num_epochs` added for the main.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/multigpu/nc_sage.py
Comment (frozenbugs) [inline examples/multigpu/nc_sage.py:1]: node_classification_sage.py

File: examples/multigpu/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+```
+
+## How to run
+
+### Node classification
+
+Run with following (available dataset: "ogbn-products", "ogbn-arxiv")
+
+```bash
+python3 nc_sage.py --dataset_name ogbn-products
```
Comment (frozenbugs) [inline examples/multigpu/README.md:16]:
```suggestion
python3 node_classification_sage.py --dataset_name ogbn-products
```

File: examples/multigpu/nc_sage.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    def forward(self, blocks, x):
+        h = x
+        for l, (layer, block) in enumerate(zip(self.layers, blocks)):
+            h = layer(block, h)
+            if l != len(self.layers) - 1:
+                h = F.relu(h)
+                h = self.dropout(h)
+        return h
+
+    def inference(self, g, device, batch_size, use_uva):
```
Comment (frozenbugs) [inline examples/multigpu/nc_sage.py:46]: Is it a good practise to embed data loader e.t.c. under a module class?

It is not related to the goal of this pr, just want to make sure all the examples in the new directory are high quality.

Comment (Rhett-Ying) [reply examples/multigpu/nc_sage.py:46]: I think it's mainly due to layerwise inference which is different from we usually do in train stage(get multiple layers in each iteration).

File: examples/multigpu/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+Run with following (available dataset: "ogbn-products", "ogbn-arxiv")
+
+```bash
+python3 node_classification_sage.py --dataset_name ogbn-products
+```
+
+#### __Results__ with default arguments
+```
+* Test Accuracy of "ogbn-products": ~0.7632
+* Test Accuracy of "ogbn-arxiv": ~0.5397
```
Comment (mufeili) [inline examples/multigpu/README.md:22]: The result for `ogbn-arxiv` is quite low. See if adding reverse edges helps.

Comment (mufeili) [reply examples/multigpu/README.md]: No. Could you try https://docs.dgl.ai/generated/dgl.to_bidirected.html#dgl.to_bidirected

File: examples/multigpu/node_classification_sage.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    return MF.accuracy(
+        torch.cat(y_hats),
+        torch.cat(ys),
+        task="multiclass",
+        num_classes=num_classes,
+    )
+
+
+def layerwise_infer(
+    proc_id, device, g, num_classes, nid, model, use_uva, batch_size=2**10
```
[the remark this answers is not in the document]
Comment (mufeili) [reply examples/multigpu/node_classification_sage.py:108]: No. It's just that I saw a different value in the script you referenced, so I ask for clarification.

File: examples/multigpu/README.md (hunk 3 above)
Comment (Rhett-Ying) [reply examples/multigpu/README.md]: `to_bidirectted` + self_loop: `Test accuracy 0.6994` How about this number?

Comment (mufeili) [reply examples/multigpu/README.md]: This appears much more reasonable. Thanks.

**Review by mufeili — APPROVED** [review]
I'm good.

File: examples/multigpu/node_classification_sage.py [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+    def forward(self, blocks, x):
+        h = x
+        for l, (layer, block) in enumerate(zip(self.layers, blocks)):
+            h = layer(block, h)
+            if l != len(self.layers) - 1:
+                h = F.relu(h)
+                h = self.dropout(h)
+        return h
+
+    def inference(self, g, device, batch_size, use_uva):
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:47]: @mufeili is it proper to wrap inference under model like this?

Comment (Rhett-Ying) [reply examples/multigpu/node_classification_sage.py:47]: @mufeili what's the best/latest practice of inference? or should we change it here?

Comment (mufeili) [reply examples/multigpu/node_classification_sage.py:47]: Sorry for the confusion. I think this is already the latest practice. Any comments? @BarclayII
